components/gitpod-protocol/src/gitpod-service.ts

```diff
@@ -280,6 +280,11 @@ export interface GitpodServer extends JsonRpcServer<GitpodClient>, AdminServer,
      * getIDToken - doesn't actually do anything, just used to authenticat/authorise
      */
     getIDToken(): Promise<void>;
+
+    /**
+     * whether this is a dedicated installation
+     */
+    isDedicatedInstallation(): Promise<boolean>;
```

Comment:
With this, how do we ensure we don't end up splitting the product in two branches - SaaS and Dedicated? We've historically avoided this (even with Self-hosted) as it creates this split brain.

Should go with it, it may be useful to make it an object such as:

```
dedicated: {
enabled: true/false,
anotherFeatureOfDedicated: true/false,
}
```
you can nest other sub-properties in there

Reply:
> With this, how do we ensure we don't end up splitting the product in two branches - SaaS and Dedicated?

I'm also on the fence with this for that same reason. We might too easily just add new stuff behind this flag without thinking through how this could work in all scenarios. @brad made the point that we are doing this a bit too much and have a ton of different flags, per feature where it really is the dedicated vs. SaaS. 
I believe we'd need to be disciplined about sparsely using this flag.

> Should go with it, it may be useful to make it an object such as:
I'd prefer to keep it a single boolean. The point of this is to reduce the amount of variance.